strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/metadata/MavenMetadataManager.java

```diff
@@ -348,23 +325,50 @@ public Metadata generateSnapshotVersioningMetadata(Path snapshotBasePath,
         return snapshotMetadata;
     }
 
-    /**
-     * Merge the existing metadata file of an artifact with the incoming new metadata.
-     *
-     * @param repository    Repository
-     * @param artifact      Artifact
-     * @param mergeMetadata Metadata
-     * @throws IOException
-     * @throws XmlPullParserException
-     */
-    public void mergeMetadata(Repository repository, Artifact artifact, Metadata mergeMetadata)
+    public void mergeAndStore(final Path metadataBasePath,
+                              final Metadata mergeMetadata)
+    {
+        doInLock(metadataBasePath, path ->
+        {
+            if (Files.exists(metadataBasePath))
+            {
+                try
+                {
+                    final Metadata metadata = readMetadata(metadataBasePath);
+                    mergeAndStore(metadataBasePath, metadata, mergeMetadata);
+                    return;
+                }
+                catch (Exception e)
+                {
+                    logger.error("Unable to merge the metadata to " + metadataBasePath + " by source metadata" +
```

Comment:
Cosmetic: It looks like there's a missing space in `" by source metadata"` and no dot before `" Continuing..."`?